packages/Tabs/src/Tabs.js

```diff
@@ -26,73 +25,96 @@ export default function Tabs(props) {
     tabHeight,
   } = props;
 
-  const indexToUse = index !== null ? index : defaultIndex;
+  const isControlled = Boolean(onClickTab);
+
+  function getIndex() {
+    if (index === null) {
+      return defaultIndex >= 0 ? defaultIndex : null;
+    }
+    return index >= 0 ? index : null;
+  }
+
+  const indexToUse = getIndex();
   const [activeIndex, setActiveIndex] = React.useState(indexToUse);
-  const [currentFocusIndex, setFocusIndex] = React.useState(indexToUse);
-  let tabListRef = React.useRef(null);
+  const [focusIndex, setFocusIndex] = React.useState(indexToUse || 0);
+  const refList = React.useRef(null);
+  const [idTabs] = React.useState(uuidv4());
 
-  React.useEffect(() => {
+  React.useLayoutEffect(() => {
     setActiveIndex(indexToUse);
-  }, [indexToUse, setActiveIndex]);
+  }, [indexToUse]);
```

Comment:
```js
const [activeIndex, setActiveIndex] = React.useState(defaultIndex >= 0 ? defaultIndex : null);
const indexToUse = isControlled : index ? activeIndex;
```

No need to have an effect here.

See https://wegalvanize.slack.com/archives/CB3QRPDJ4/p1616523810022900?thread_ts=1616522183.021000&cid=CB3QRPDJ4